sunpy/net/scraper.py

```diff
@@ -3,101 +3,175 @@
     regex : `bool`
         Set to `True` if parts of the pattern uses regexp symbols.
         This only works for the filename part of the pattern rather than the full url.
         Be careful that periods ``.`` matches any character and therefore it's better to escape them.
         If regexp is used, other ``kwargs`` are ignored and string replacement is
         not possible. Default is `False`.
+        Deprecated in favor of `format`. Default is `False`.
+    format : `str`
+        A string containing the url with the date and other information to be
+        extracted encoded as ``parse`` formats, and any other ``kwargs`` parameters
+        as a string format, the former represented using double curly-brackets
+        to differentiate from the latter.
+        The accepted parse representations for datetime values are as given in ``PARSE_TIME_CONVERSIONS``.
+        This can also be a uri to a local file patterns. Default is `None`.
+    kwargs : `dict`
+        A dictionary containing the values to be replaced in the pattern.
+        Will be ignored if ``regex`` is `True`.
 
     Attributes
     ----------
     pattern : `str`
-        A converted string with the kwargs.
+        The pattern with the parse format.
+    datetime_pattern : `str`
+        The parse pattern in the datetime format.
     now : `datetime.datetime`
         The pattern with the actual date.
+        This is not checking if there is an existent file, but just how the ``pattern`` looks with the current time.
 
     Examples
     --------
     >>> from sunpy.net import Scraper
-    >>> pattern = ('http://proba2.oma.be/{instrument}/data/bsd/%Y/%m/%d/'
-    ...            '{instrument}_lv1_%Y%m%d_%H%M%S.fits')
-    >>> swap = Scraper(pattern, instrument='swap')
+    >>>
+    >>> pattern = ('http://proba2.oma.be/{instrument}/data/bsd/{{year:4d}}/{{month:2d}}/{{day:2d}}/'
+    ...            '{instrument}_lv1_{{year:4d}}{{month:2d}}{{day:2d}}_{{hour:2d}}{{month:2d}}{{second:2d}}.fits')
+    >>> swap = Scraper(format=pattern, instrument='swap')
+    >>>
     >>> print(swap.pattern)
-    http://proba2.oma.be/swap/data/bsd/%Y/%m/%d/swap_lv1_%Y%m%d_%H%M%S.fits
+    http://proba2.oma.be/swap/data/bsd/{year:4d}/{month:2d}/{day:2d}/swap_lv1_{year:4d}{month:2d}{day:2d}_{hour:2d}{month:2d}{second:2d}.fits
+    >>>
+    >>> print(swap.datetime_pattern)
+    http://proba2.oma.be/swap/data/bsd/%Y/%m/%d/swap_lv1_%Y%m%d_%H%m%S.fits
+    >>>
     >>> print(swap.now)  # doctest: +SKIP
     http://proba2.oma.be/swap/data/bsd/2022/12/21/swap_lv1_20221221_112433.fits
-
-    Notes
-    -----
-    The ``now`` attribute does not return an existent file, but just how the
-    pattern looks with the actual time.
     """
-
-    def __init__(self, pattern, regex=False, **kwargs):
-        if regex:
+    @deprecated_renamed_argument("pattern", None, since="6.0", message=DEPRECATED_MESSAGE)
+    @deprecated_renamed_argument("regex", None, since="6.0", message=DEPRECATED_MESSAGE)
```

Comment:
```suggestion
    @deprecated_renamed_argument("pattern", None, since="6.1", message=DEPRECATED_MESSAGE)
    @deprecated_renamed_argument("regex", None, since="6.1", message=DEPRECATED_MESSAGE)
```